### ISSUE TITLE
NPE within Auction House

### ISSUE THREAD

**Issue by StrangeOne101** [SkyblockerMod/Skyblocker#858]
### What happened?

Trying to view a bid on a certain item in the auction house causes this NPE. This started after I clicked a cake bag within an auction house, and is persistent even over relogs AND restarts of the client

### Log output

```
[16:38:05] [Render thread/ERROR]: Error executing task on Client
net.minecraft.class_148: Main thread packet handler
	at net.minecraft.class_2600.method_11072(class_2600.java:33) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_18859(class_1255.java:156) ~[client-intermediary.jar:?]
	at net.minecraft.class_4093.method_18859(class_4093.java:23) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:130) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_5383(class_1255.java:115) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1283) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:888) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:265) ~[minecraft-1.20.4-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.15.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.15.11.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:100) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129) ~[NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) ~[NewLaunch.jar:?]
Caused by: java.lang.NullPointerException: Cannot invoke "net.minecraft.class_7842.method_25355(net.minecraft.class_2561)" because "this.infoTextWidget" is null
	at de.hysky.skyblocker.skyblock.auction.AuctionViewScreen.changeState(AuctionViewScreen.java:99) ~[skyblocker-1.20.2+1.20.4.jar:?]
	at de.hysky.skyblocker.skyblock.auction.AuctionViewScreen.onSlotChange(AuctionViewScreen.java:204) ~[skyblocker-1.20.2+1.20.4.jar:?]
	at de.hysky.skyblocker.skyblock.auction.AuctionViewScreen.onSlotChange(AuctionViewScreen.java:30) ~[skyblocker-1.20.2+1.20.4.jar:?]
	at de.hysky.skyblocker.utils.render.gui.AbstractCustomHypixelGUI.method_7635(AbstractCustomHypixelGUI.java:48) ~[skyblocker-1.20.2+1.20.4.jar:?]
	at net.minecraft.class_1703.method_34246(class_1703.java:219) ~[client-intermediary.jar:?]
	at net.minecraft.class_1703.method_7623(class_1703.java:176) ~[client-intermediary.jar:?]
	at net.minecraft.class_1703.method_7609(class_1703.java:589) ~[client-intermediary.jar:?]
	at de.hysky.skyblocker.skyblock.auction.AuctionHouseScreenHandler.method_7619(AuctionHouseScreenHandler.java:56) ~[skyblocker-1.20.2+1.20.4.jar:?]
	at net.minecraft.class_634.method_11109(class_634.java:1217) ~[client-intermediary.jar:?]
	at net.minecraft.class_2653.method_11451(class_2653.java:41) ~[client-intermediary.jar:?]
	at net.minecraft.class_2653.method_11054(class_2653.java:8) ~[client-intermediary.jar:?]
	at net.minecraft.class_2600.method_11072(class_2600.java:24) ~[client-intermediary.jar:?]
	... 13 more
[16:38:06] [Render thread/WARN]: Ignoring click in mismatching container. Click in 2, player has 3.
```

### Minecraft Version

1.21.0

### Skyblocker Version

v1.20.2+1.20.4

**Comment by kevinthegreat1** [comment]
I wasn't able to figure out the bug, can you try [this debug build](https://github.com/SkyblockerMod/Skyblocker/actions/runs/10035596947/artifacts/1724516799) and post the full logs you get?

**Comment by StrangeOne101** [comment]
I couldn't replicate it with the debug build. I'm not sure if anything else changes between 1.20.4 and the 1.21 debug build, but I certainly couldn't encounter it again even after all my testing

Perhaps the issue should be closed as a fixed legacy issue then

**Comment by kevinthegreat1** [comment]
It looks like fancy auction house hasn't been changed since it was added. It may also be an incompatibility so you can see if it happens with just skyblocker.

**Comment by viciscat** [comment]
this bug has been fixed in more recent versions of the mod 🙃 #689

**Comment by StrangeOne101** [comment]
Alright perfect. Thanks. Issue resolved then
